nullaway/src/main/java/com/uber/nullaway/handlers/MethodAnalysisContext.java

```diff
@@ -0,0 +1,43 @@
+package com.uber.nullaway.handlers;
+
+import com.google.errorprone.VisitorState;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.uber.nullaway.NullAway;
+
+public class MethodAnalysisContext {
```

Comment:
If we were able to use Java 17 features, I think a record would be perfect for this use case.  Since we cannot use records, I think several changes would help here:

* This type should be immutable (`final` fields and no setters).
* It should implement equals, hashCode, and toString. 
* Like records, I would avoid the `get` prefix on getter methods; so just have them be `analysis()`, `state()`, and `methodSymbol()`.  
* The type also needs top-level Javadoc.

Reply:
Thank you very much for the feedback. I can implement your suggestions in the coming days :)

Reply:
done